### ISSUE TITLE
Unable to use PEM key

### ISSUE THREAD

**Issue by enyo** [appsup-dart/x509#5]
I'm using a PEM key from apple and I get this:

`Unknown algoritmh ecPublicKey`

```dart
print(parsePem('''-----BEGIN PRIVATE KEY-----
[...]
[...]
[...]
[...]
-----END PRIVATE KEY-----'''));
```

```
% dart bin/test.dart
Unhandled exception:
UnimplementedError: Unknown algoritmh ecPublicKey
#0      keyPairFromAsn1 (package:x509/src/util.dart:142:3)
#1      new PrivateKeyInfo.fromAsn1 (package:x509/src/x509_base.dart:168:9)
#2      _parseDer (package:x509/src/x509_base.dart:226:29)
#3      parsePem.<anonymous closure> (package:x509/src/x509_base.dart:269:11)
#4      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:165:25)
#5      _iterablePartsToStrings (dart:collection/iterable.dart:318:13)
#6      IterableBase.iterableToShortString (dart:collection/iterable.dart:238:7)
#7      Iterable.toString (dart:core/iterable.dart:672:37)
#8      print (dart:core/print.dart:9:24)
#9      main (file:///Users/enyo/tmp/aa112/dart/bin/test.dart:5:3)
#10     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:301:19)
#11     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

**Comment by JordySinke** [comment]
Did you find a solution for this? I have the same problem parsing a .p8 key from Apple.

**Comment by enyo** [comment]
@JordySinke no... unfortunately I didn't. I am now installing ruby and am using a ruby script to create the JWT I need. Needless to say, that's rather suboptimal.

I also tried converting the key to JSON format, and got it to work, but the signature that I got out of it wasn't accepted by apple and I have no idea why.

If you find a way to do this in dart I'd be very thankful if you would share your approach.
